mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java

```diff
@@ -636,11 +638,13 @@ private ResourceOverview getResourceOverview() {
         return this.executorStateManager.getResourceOverview();
     }
 
-    private TaskExecutorStatus getTaskExecutorStatus(TaskExecutorID taskExecutorID) {
+    @Nonnull
+    private Object getTaskExecutorStatus(TaskExecutorID taskExecutorID) throws ResourceNotFoundException {
```

Comment:
nit: It's not throwing the exception.

Reply:
After your change, the only caller for this method is within the Akka context. Thus, you might as well change the signature to be reflective of that. 
```
private void onGetTaskExecutorStatus() {
  // send the message directly here. 
  getSender().tell();
}
```